### DISCUSSION TITLE
Errore connessione openvpn Client

### DISCUSSION THREAD

**Discussion by dellannajw** [indomus/forum#2663] [HUB: Home Assistant]
Buongiorno a tutti. Ho seguito pedissequamente la guida ma quando tento di avviare l’add-on OpenVPN Client da HA il log genera questi messaggi:

jq: error (at <stdin>:2): Cannot iterate over null (null)

[2023.10.19-15:27:30 | 1 : run.sh ] Wait until the user uploads the files.

[2023.10.19-15:27:30 | 1 : run.sh ] Start the web server.

[2023.10.19-15:27:30 | 1 : run.sh ] File /data//client.crt not found

[uWSGI] getting INI configuration from /openvpnclient/venv/share/openvpnclient/openvpnclient.ini

* Starting uWSGI 2.0.22 (64bit) on [Thu Oct 19 15:27:30 2023] *

compiled with version: 12.2.1 20220924 on 19 October 2023 13:24:45

os: Linux-6.1.21-v8 #1 SMP Fri Oct 13 11:30:01 UTC 2023

nodename: local-openvpn-client

machine: aarch64

clock source: unix

pcre jit disabled

detected number of CPU cores: 4

current working directory: /openvpnclient/venv

detected binary path: /openvpnclient/venv/bin/uwsgi

uWSGI running as root, you can use --uid/--gid/--chroot options

* WARNING: you are running uWSGI as root !!! (use the --uid flag) *

your memory page size is 4096 bytes

detected max file descriptor number: 1073741816

lock engine: pthread robust mutexes

thunder lock: disabled (you can enable it with --thunder-lock)

uwsgi socket 0 bound to TCP address 0.0.0.0:8090 fd 3

uWSGI running as root, you can use --uid/--gid/--chroot options

* WARNING: you are running uWSGI as root !!! (use the --uid flag) *

Python version: 3.11.6 (main, Oct 4 2023, 06:22:18) [GCC 12.2.1 20220924]

PEP 405 virtualenv detected: /openvpnclient/venv

Set PythonHome to /openvpnclient/venv

* Python threads support is disabled. You can enable it with --enable-threads *

Python main interpreter initialized at 0x7f8f4c22c0

uWSGI running as root, you can use --uid/--gid/--chroot options

* WARNING: you are running uWSGI as root !!! (use the --uid flag) *

your server socket listen backlog is limited to 100 connections

your mercy for graceful operations on workers is 60 seconds

mapped 437424 bytes (427 KB) for 5 cores

* Operational MODE: preforking *

WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x7f8f4c22c0 pid: 19 (default app)

uWSGI running as root, you can use --uid/--gid/--chroot options

* WARNING: you are running uWSGI as root !!! (use the --uid flag) *

* uWSGI is running in multiple interpreter mode *

spawned uWSGI master process (pid: 19)

spawned uWSGI worker 1 (pid: 21, cores: 1)

spawned uWSGI worker 2 (pid: 22, cores: 1)

spawned uWSGI worker 3 (pid: 23, cores: 1)

spawned uWSGI worker 4 (pid: 24, cores: 1)

spawned uWSGI worker 5 (pid: 25, cores: 1)

[2023.10.19-15:33:41 | 1 : run.sh ] File /data//client.crt not found

In share ci sono i 2 file:

- client.ovpn (modificato con l’aggiunta in cima di “askpass /share/nome-utente.pwd”

- nome-utente.pwd (con la password in chiaro per connettersi all’OpenVPN Server, che nel mio caso è una VM in GCP)

Premetto che da client OpenVPN Android la connessione VPN usando lo stesso file client.ovpn funziona regolarmente

Aiutatemi, please🙏

**Comment by dellannajw** [comment]
Ragazzi non ne vengo fuori. Mi aiutate a capire dove sbaglio?

**Comment by gomez85** [comment]
> Buongiorno a tutti. Ho seguito pedissequamente la guida ma quando tento di avviare l’add-on OpenVPN Client da HA il log genera questi messaggi:
> 
> jq: error (at :2): Cannot iterate over null (null)
> 
> [2023.10.19-15:27:30 | 1 : run.sh ] Wait until the user uploads the files.
> 
> [2023.10.19-15:27:30 | 1 : run.sh ] Start the web server.
> 
> [2023.10.19-15:27:30 | 1 : run.sh ] File /data//client.crt not found
> 
> [uWSGI] getting INI configuration from /openvpnclient/venv/share/openvpnclient/openvpnclient.ini
> 
> * Starting uWSGI 2.0.22 (64bit) on [Thu Oct 19 15:27:30 2023] *
> 
> compiled with version: 12.2.1 20220924 on 19 October 2023 13:24:45
> 
> os: Linux-6.1.21-v8 #1 SMP Fri Oct 13 11:30:01 UTC 2023
> 
> nodename: local-openvpn-client
> 
> machine: aarch64
> 
> clock source: unix
> 
> pcre jit disabled
> 
> detected number of CPU cores: 4
> 
> current working directory: /openvpnclient/venv
> 
> detected binary path: /openvpnclient/venv/bin/uwsgi
> 
> uWSGI running as root, you can use --uid/--gid/--chroot options
> 
> * WARNING: you are running uWSGI as root !!! (use the --uid flag) *
> 
> your memory page size is 4096 bytes
> 
> detected max file descriptor number: 1073741816
> 
> lock engine: pthread robust mutexes
> 
> thunder lock: disabled (you can enable it with --thunder-lock)
> 
> uwsgi socket 0 bound to TCP address 0.0.0.0:8090 fd 3
> 
> uWSGI running as root, you can use --uid/--gid/--chroot options
> 
> * WARNING: you are running uWSGI as root !!! (use the --uid flag) *
> 
> Python version: 3.11.6 (main, Oct 4 2023, 06:22:18) [GCC 12.2.1 20220924]
> 
> PEP 405 virtualenv detected: /openvpnclient/venv
> 
> Set PythonHome to /openvpnclient/venv
> 
> * Python threads support is disabled. You can enable it with --enable-threads *
> 
> Python main interpreter initialized at 0x7f8f4c22c0
> 
> uWSGI running as root, you can use --uid/--gid/--chroot options
> 
> * WARNING: you are running uWSGI as root !!! (use the --uid flag) *
> 
> your server socket listen backlog is limited to 100 connections
> 
> your mercy for graceful operations on workers is 60 seconds
> 
> mapped 437424 bytes (427 KB) for 5 cores
> 
> * Operational MODE: preforking *
> 
> WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x7f8f4c22c0 pid: 19 (default app)
> 
> uWSGI running as root, you can use --uid/--gid/--chroot options
> 
> * WARNING: you are running uWSGI as root !!! (use the --uid flag) *
> * uWSGI is running in multiple interpreter mode *
> 
> spawned uWSGI master process (pid: 19)
> 
> spawned uWSGI worker 1 (pid: 21, cores: 1)
> 
> spawned uWSGI worker 2 (pid: 22, cores: 1)
> 
> spawned uWSGI worker 3 (pid: 23, cores: 1)
> 
> spawned uWSGI worker 4 (pid: 24, cores: 1)
> 
> spawned uWSGI worker 5 (pid: 25, cores: 1)
> 
> [2023.10.19-15:33:41 | 1 : run.sh ] File /data//client.crt not found
> 
> In share ci sono i 2 file:
> 
> * client.ovpn (modificato con l’aggiunta in cima di “askpass /share/nome-utente.pwd”
> * nome-utente.pwd (con la password in chiaro per connettersi all’OpenVPN Server, che nel mio caso è una VM in GCP)
> 
> Premetto che da client OpenVPN Android la connessione VPN usando lo stesso file client.ovpn funziona regolarmente
> 
> Aiutatemi, please🙏

Ciao,
ho il tuo stesso problema. Per caso sei riuscito a risolvere? come?

Grazie mille

**Comment by dellannajw** [comment]
No purtroppo. Alla fine ho rinunciato anche perché non sono più stato sul
posto per fare ulteriori prove
